tinytodo/src/context.rs

```diff
@@ -355,27 +356,101 @@ impl AppContext {
+                            // found template in new policy set
+                            match p.template_links() {
+                                None => error!("Error when reloading policies: Template with no matching links"),
+                                Some(vals) => {
+                                    // link against new template, using the same policy ID as the old one
+                                    updated = true;
+                                    new_policies.link(tid.clone(), p.id().clone(), vals)?
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        // no error during relinking
+        if !err {
+            // check that re-linked policies validate properly
+            if updated {
+                let validator = Validator::new(self.schema.clone());
+                let output = validator.validate(&new_policies, ValidationMode::default());
+                if !output.validation_passed() {
+                    for e in output.validation_errors() {
+                        error!("Error validating linked policies: {e}")
+                    }
+                } else {
+                    self.policies = new_policies;
+                    info!("Reloaded policy set")
+                }
+            }
+        };
         Ok(AppResponse::Unit(()))
     }
 
     fn add_share(&mut self, r: AddShare) -> Result<AppResponse> {
         self.is_authorized(&r.uid, &*ACTION_EDIT_SHARE, &r.list)?;
-        let list = self.entities.get_list(&r.list)?;
-        let team_uid = list.get_team(r.role).clone();
-        let target_entity = self.entities.get_user_or_team_mut(&r.share_with)?;
-        target_entity.insert_parent(team_uid);
+        // Confirm that the identified list and sharer are known
+        let _list = self.entities.get_list(&r.list)?;
+        let _target_entity = self.entities.get_user_or_team_mut(&r.share_with)?;
+        // Link a template to register the new permission
+        let (tid, pid_prefix) = match r.role {
+            ShareRole::Reader => (PolicyId::from_str("reader-template")?, "reader"),
+            ShareRole::Editor => (PolicyId::from_str("editor-template")?, "editor"),
+        };
+        // Construct template linking values
+        let target_euid: &cedar_policy::EntityUid = r.share_with.as_ref();
+        let list_euid: &cedar_policy::EntityUid = r.list.as_ref();
+        let env: HashMap<SlotId, cedar_policy::EntityUid> = [
+            (SlotId::principal(), target_euid.clone()),
+            (SlotId::resource(), list_euid.clone()),
+        ]
+        .into_iter()
+        .collect();
+        // Construct policy ID; assumes no policy in the set has it already
+        let target_eid = target_euid.id();
+        let list_eid = list_euid.id();
+        let pid = PolicyId::from_str(&format!("{pid_prefix}[{target_eid}][{list_eid}]"))?;
```

Comment:
Should use a UUID here

Reply:
I think it is. The list ID is under the app's control, so the last part of the ID is always `[N]` where `N` is a number, and the first part always `reader[` or `editor[`. The stuff in the middle is always the target team's ID with a trailing `]`. Is this reasoning flawed?

Reply:
I think you're right

Reply:
I'll add a comment that makes this clear.